xds/internal/xdsclient/xdsresource/unmarshal_rds.go

```diff
@@ -313,29 +314,23 @@ func routesProtoToSlice(routes []*v3routepb.Route, csps map[string]clusterspecif
 				route.WeightedClusters[a.Cluster] = WeightedCluster{Weight: 1}
 			case *v3routepb.RouteAction_WeightedClusters:
 				wcs := a.WeightedClusters
-				var totalWeight uint32
+				var totalWeight uint64
 				for _, c := range wcs.Clusters {
 					w := c.GetWeight().GetValue()
 					if w == 0 {
 						continue
 					}
+					if (totalWeight + uint64(w)) > math.MaxUint32 {
+						return nil, nil, fmt.Errorf("total weight of clusters exceed MaxUint32")
```

Comment:
Nit: s/exceed/exceeds/
Also, add an `xds:` prefix to the error string.

Reply:
done.